solr/core/src/java/org/apache/solr/response/SmileResponseWriter.java

```diff
@@ -24,16 +24,27 @@
 import java.math.BigInteger;
 import org.apache.solr.request.SolrQueryRequest;
 
-public class SmileResponseWriter extends BinaryResponseWriter {
+/**
+ * A Smile formatting {@link QueryResponseWriter}.
+ *
+ * @see <a href="https://github.com/FasterXML/smile-format-specification">Smile</a>
+ */
+public class SmileResponseWriter implements QueryResponseWriter {
 
   @Override
-  public void write(OutputStream out, SolrQueryRequest request, SolrQueryResponse response)
+  public void write(
+      OutputStream out, SolrQueryRequest request, SolrQueryResponse response, String contentType)
       throws IOException {
     try (SmileWriter sw = new SmileWriter(out, request, response)) {
       sw.writeResponse();
     }
   }
 
+  @Override
+  public String getContentType(SolrQueryRequest request, SolrQueryResponse response) {
```

Comment:
Smile was wrongly subclassing BinaryResponseWriter (JavaBin) and thus inheriting its content type.  It was an innocent mistake enabled by BinaryResponseWriter's bad name; should say "JavaBin" in there.